docs/about-core-concepts/architecture-guide.rst

```diff
@@ -10,7 +10,7 @@ USGS's Landsat Collection 2, Planet and a wide range of other organisations.
 The ODC can index from STAC, although this process is not fully integrated, it is
```

Comment:
I wonder whether it's worth explaining more clearly in this section that the Open Data Cube relies on both data (e.g. COGS) and metadata (e.g. STAC), perhaps with a reference to where people can read more about the metadata types. Then, below, we can make it clear that you need some form of metadata to index data into the database.